subsys/bluetooth/services/fast_pair/CMakeLists.txt

```diff
@@ -30,47 +30,49 @@ if(CONFIG_BT_FAST_PAIR_STORAGE)
   target_link_libraries(fast_pair PRIVATE fp_storage)
 endif()
 
-set(FP_PARTITION_NAME bt_fast_pair)
+if(NOT SYSBUILD)
+  set(FP_PARTITION_NAME bt_fast_pair)
```

Comment:
I wonder if we could avoid code duplication by e.g. including `fast_pair_hex.cmake` here? I would also suggest to fail build (or at least emit CMake warning) if Fast Pair is enabled in the application and provisioning data is not properly generated.

Reply:
I tested locally and `samples/bluetooth/fast_pair/input_device` built with `west build --sysbuild -p -b nrf52840dk_nrf52840` crashes on boot because of invalid provisioning data.

Reply:
That is because by itself, this PR does nothing :) you would need to cherry-pick the commits on top of https://github.com/nrfconnect/sdk-nrf/pull/12918

Reply:
yes, I built this and tested it on an nrf52840dk, the mouse was moving when buttons were pressed. `cmake -GNinja -DBOARD=nrf52840dk_nrf52840 -DCONF_FILE=prj_fast_pair.conf -DFP_MODEL_ID=0x8E717D DFP_ANTI_SPOOFING_KEY=dZxFzP7X9CcfLPC0apyRkmgsh3n2EbWo9NFNXfVuxAM= -DAPP_DIR=.. -DSB_CONFIG_BOOTLOADER_MCUBOOT=y /tmp/bb/zephyr/share/sysbuild/`

Reply:
Do you provide additional changes that are not available on these PRs?

Reply:
> I wonder if we could avoid code duplication by e.g. including fast_pair_hex.cmake here?

No